crypto/rand/drbg_lib.c

```diff
@@ -847,63 +847,14 @@ static void drbg_delete_thread_state(void *arg)
     RAND_DRBG_free(drbg);
 }
 
-/* Implements the default OpenSSL RAND_bytes() method */
-static int drbg_bytes(unsigned char *out, int count)
-{
-    int ret;
-    RAND_DRBG *drbg = RAND_DRBG_get0_public();
-
-    if (drbg == NULL)
-        return 0;
-
-    ret = RAND_DRBG_bytes(drbg, out, count);
-
-    return ret;
-}
-
-/* Implements the default OpenSSL RAND_add() method */
-static int drbg_add(const void *buf, int num, double randomness)
-{
-    RAND_DRBG *drbg = RAND_DRBG_get0_master();
-
-    if (drbg == NULL || num <= 0)
-        return 0;
-
-    return EVP_RAND_reseed(drbg->rand, 0, NULL, 0, buf, num);
-}
-
-/* Implements the default OpenSSL RAND_seed() method */
-static int drbg_seed(const void *buf, int num)
-{
-    return drbg_add(buf, num, num);
-}
-
-/* Implements the default OpenSSL RAND_status() method */
-static int drbg_status(void)
-{
-    int ret;
-    RAND_DRBG *drbg = RAND_DRBG_get0_master();
-
-    if (drbg == NULL)
-        return 0;
-
-    ret = EVP_RAND_state(drbg->rand) == EVP_RAND_STATE_READY ? 1 : 0;
-    return ret;
-}
-
-int RAND_DRBG_verify_zeroization(RAND_DRBG *drbg)
-{
-    return EVP_RAND_verify_zeroization(drbg->rand);
-}
-
 /*
  * Get the primary DRBG.
  * Returns pointer to the DRBG on success, NULL on failure.
  *
  */
-RAND_DRBG *OPENSSL_CTX_get0_primary_drbg(OPENSSL_CTX *ctx)
+RAND_DRBG *RAND_DRBG_get0_master(void)
```

Comment:
Also the RAND_DRBG API should use the term 'primary'. 

```suggestion
RAND_DRBG *RAND_DRBG_get0_primary(void)
```

To retain compatibility, all you need is to add a compatibility define to <openssl/rand_drbg.h>.

```C
#define RAND_DRBG_get0_master  RAND_DRBG_get0_primary
```

Reply:
Is this really required in a legacy API that **will** be removed as soon as possible?  The old name must remain until deprecation, it didn't seem worthwhile creating a new name.

Reply:
I think we agree that removing the word "master" from our source code did not have a technical reason. It is a symbolic act to affirm our open and welcoming attitude to the community. Keeping the public symbol in the library thwarts this intention, because "as soon as possible" means that this exported symbol will remain there for at least for the next five years. So yes, I think it's worth the extra effort to minimize the usage of the unwanted word.

Reply:
Okay, done.